lib/hooks/session/index.js

```diff
@@ -32,6 +32,13 @@ module.exports = function(app) {
         // By default, disable session for requests to paths that look like static assets.
         isSessionDisabled: function (req){
           return !!req.path.match(req._sails.LOOKS_LIKE_ASSET_RX);
+        },
+        // Added overrideable function for constructing the session store based on https://github.com/balderdashy/sails/pull/7172
+        // • `sessionConfig` is what Sails has loaded for sails.config.session
+        // • `configuredSessionAdapter` is the package like what you get from doing require('connect-mongo')
+        // • `expressSessionFromSailsCore` is the version of express-session used by Sails core session hook, provided for your convenience
+        handleConstructingSessionStore: (sessionConfig, configuredSessionAdapter, expressSessionFromSailsCore) => {
+          return new configuredSessionAdapter(sessionConfig);
```

Comment:
At the moment this will support `connect-mongo` v4.5 I can change it to support `connect-mongo` v2 by changing the body to:
```
var CustomStore = configuredSessionAdapter(expressSessionFromSailsCore);
return new CustomStore(sessionConfig);
```